module/enrichers.mjs

```diff
@@ -345,6 +357,75 @@ async function enrichDamage(config, label, options) {
   return span;
 }
 
+/* -------------------------------------------- */
+/*  Embed Enrichers                             */
+/* -------------------------------------------- */
+
+const MAX_EMBED_DEPTH = 5;
+
+/* -------------------------------------------- */
+
+/**
+ * Parse the enriched embed and provide the appropriate content.
+ * @param {object} config              Configuration data.
+ * @param {string} [label]             Optional label to replace default caption/text.
+ * @param {EnrichmentOptions} options  Options provided to customize text enrichment.
+ * @returns {HTMLElement|null}         An HTML link if the check could be built, otherwise null.
+ */
+async function enrichEmbed(config, label, options) {
+  options._embedDepth ??= 0;
+  if ( options._embedDepth > MAX_EMBED_DEPTH ) {
+    console.warn(
+      `Embed enrichers are restricted to ${MAX_EMBED_DEPTH} levels deep. ${config._input} cannot be enriched fully.`
+    );
+    return null;
+  }
+
+  for ( const value of config.values ) {
+    if ( config.uuid ) break;
+    try {
+      const parsed = foundry.utils.parseUuid(value);
+      if ( parsed.documentId ) config.uuid = value;
+    } catch(err) {}
+  }
+
+  config.doc = await fromUuid(config.uuid, { relative: options.relativeTo });
+  if ( config.doc instanceof JournalEntryPage ) {
+    switch ( config.doc.type ) {
+      case "image": return embedImagePage(config, label, options);
+      case "text":
+      case "rule": return embedTextPage(config, label, options);
+    }
+  }
+  else if ( config.doc instanceof RollTable ) return embedRollTable(config, label, options);
+  else if ( (config.doc instanceof Actor) || (config.doc instanceof Item) ) {
+    return embedDocument(config, label, options);
+  }
+  return null;
+}
+
+/* -------------------------------------------- */
+
+/**
+ * Embed an Actor or Item document with a description.
+ * @param {object} config              Configuration data.
+ * @param {string} [label]             Optional label to replace the default caption.
+ * @param {EnrichmentOptions} options  Options provided to customize text enrichment.
+ * @returns {HTMLElement|null}         An HTML figure containing the image, caption from the image page or a custom
+ *                                     caption, and a link to the source if it could be built, otherwise null.
+ */
+async function embedDocument(config, label, options) {
+  options = { ...options, _embedDepth: options._embedDepth + 1, relativeTo: config.doc };
+  config.inline ??= config.values.includes("inline");
+
+  const keyPath = (config.doc instanceof Actor) ? "system.details.biography.value" : "system.description.value";
+  const description = foundry.utils.getProperty(config.doc, keyPath);
+  if ( description === undefined ) return null;
```

Comment:
Is it desirable that a blank description be rendered as such? Or should this be:

```suggestion
  if ( !description ) return null;
```

Reply:
If we go this way then a blank enricher will render as the original enricher text. I figured it we better to have a blank space for a blank description rather than showing that and leave the enricher text fallback as an option for people trying to embed documents that don't support this format.